src/util/union_find.cpp

```diff
@@ -110,7 +110,7 @@ unsigned_union_find::size_type unsigned_union_find::get_other(size_type a)
   check_index(a);
   a=find(a);
 
-  assert(nodes[a].count>=2);
+  INVARIANT(nodes[a].count>=2, ""); // review suitable statement
```

Comment:
More or less a precondition... basically you can't find another node if the node is already isolated.